x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -86,16 +85,14 @@
      * Information about the index against which the SQL is being analyzed.
      */
     private final IndexResolution indexResolution;
-    /**
-     * Time zone in which we're executing this SQL. It is attached to functions
-     * that deal with date and time.
-     */
-    private final TimeZone timeZone;
 
-    public Analyzer(FunctionRegistry functionRegistry, IndexResolution results, TimeZone timeZone) {
+
+    private final FunctionContext context;
+
+    public Analyzer(FunctionRegistry functionRegistry, IndexResolution results, FunctionContext context) {
         this.functionRegistry = functionRegistry;
         this.indexResolution = results;
-        this.timeZone = timeZone;
+        this.context = context;
```

Comment:
Previously a bunch of classes passed a "default" tz. To make dayName with locale work it became necessary to support a "default" locale, so i replaced tz with a context that holds both "defaults".